### PR TITLE
ramips: add support for ELECOM WRC-1167GSxx devices

### PR DESCRIPTION
**Pull request by musashino205**
- add support for ELECOM WRC-1167GS2-B
- add support for ELECOM WRC-1167GST2

### CODE REVIEW

**Comment by adrianschmutzler** [comment]
Hmm, I'm not sure whether that redistribution of stuff defeats the purpose of the DTSI. I don't have a better idea yet, let me think about it ...

**Comment by musashino205** [comment]
> Hmm, I'm not sure whether that redistribution of stuff defeats the purpose of the DTSI. I don't have a better idea yet, let me think about it ...

OK, thank you.
I've also prepared another version that creates a new dtsi for WRC-1167GS2-B and WRC-1167GST2:
[Comparing openwrt:master...musashino-build:devadd/wrc-1167gsxx · openwrt/openwrt](https://github.com/openwrt/openwrt/compare/master...musashino-build:devadd/wrc-1167gsxx)

**Comment by adrianschmutzler** [comment]
Though it's really a matter of taste here, I'd prefer the alternate version you provided in your most recent post.

Note that on ramips it's actually usual to include the unique model name in the LED labels, so you should move the LEDs (and I'd prefer the whole leds block here) to the DTS files.

I'd also like the label-mac-address alias moved to the DTS files, as the addresses are defined there.

Why is there only one WiFi but two WiFi LEDs?

Does the wrc-1167gst2 really have MAC address in the calibration data?
And you are really sure about those strange 0xfffx addresses for the other device?

**Comment by musashino205** [comment]
I see, I'll update this PR.

> Why is there only one WiFi but two WiFi LEDs?

Both models have a MT7615D and it provides 2.4/5 GHz wifi by single chip. The setup for DBDC (Dual-Band-Dual-Concurrent) mode in device tree is not supported in mt76 driver, so the mode cannot be configured in the dts and the wifi node is temporary state.

> Does the wrc-1167gst2 really have MAC address in the calibration data?
And you are really sure about those strange 0xfffx addresses for the other device?

#### WRC-1167GS2-B
```
root@OpenWrt:/# cat /proc/mtd
dev:    size   erasesize  name
mtd0: 00030000 00010000 "u-boot"
mtd1: 00010000 00010000 "u-boot-env"
mtd2: 00010000 00010000 "factory"
mtd3: 00b00000 00010000 "firmware"
mtd4: 002375a3 00010000 "kernel"
mtd5: 008c8a5d 00010000 "rootfs"
mtd6: 00630000 00010000 "rootfs_data"
mtd7: 00380000 00010000 "tm_pattern"
mtd8: 00080000 00010000 "tm_key"
mtd9: 00030000 00010000 "nvram"
mtd10: 00080000 00010000 "user_data"
root@OpenWrt:/# hexdump -n 32 -C /dev/mtd2
00000000  15 76 a0 00 04 ab 18 58  2f 16 15 76 c3 14 00 80  |.v.....X/..v....|
00000010  02 00 c3 14 15 76 03 22  ff ff 23 04 0d 02 8f 02  |.....v."..#.....|
00000020
root@OpenWrt:/# hexdump -n 16 -C -s $((0xe000)) /dev/mtd2
0000e000  ff ff ff ff ff ff ff ff  ff ff ff ff ff ff ff ff  |................|
0000e010
root@OpenWrt:/# hexdump -n 16 -C -s $((0xfff0)) /dev/mtd2
0000fff0  ff ff ff ff 04 ab 18 58  2f 13 04 ab 18 58 2f 14  |.......X/....X/.|
00010000
```

#### WRC-1167GST2
```
root@OpenWrt:/# cat /proc/mtd
dev:    size   erasesize  name
mtd0: 00030000 00010000 "u-boot"
mtd1: 00010000 00010000 "u-boot-env"
mtd2: 00010000 00010000 "factory"
mtd3: 01800000 00010000 "firmware"
mtd4: 0023752c 00010000 "kernel"
mtd5: 015c8ad4 00010000 "rootfs"
mtd6: 01330000 00010000 "rootfs_data"
mtd7: 00400000 00010000 "tm_pattern"
mtd8: 00100000 00010000 "tm_key"
mtd9: 000b0000 00010000 "nvram"
mtd10: 00200000 00010000 "user_data"
root@OpenWrt:/# hexdump -n 32 -C /dev/mtd2
00000000  15 76 a0 00 00 11 22 33  44 50 15 76 c3 14 00 80  |.v...."3DP.v....|
00000010  02 00 c3 14 15 76 03 22  ff ff 23 04 0d 02 8f 02  |.....v."..#.....|
00000020
root@OpenWrt:/# hexdump -n 16 -C -s $((0xe000)) /dev/mtd2
0000e000  04 ab 18 1e ef 07 04 ab  18 1e ef 08 ff ff ff ff  |................|
0000e010
root@OpenWrt:/# hexdump -n 16 -C -s $((0xfff0)) /dev/mtd2
0000fff0  ff ff ff ff ff ff ff ff  ff ff ff ff ff ff ff ff  |................|
00010000
```

**Comment by musashino205** [comment]
UPDATE:
- switch to another version
- move ```leds``` node to each device dts files
- move ```label-mac-device``` alias to each device dts files

File: target/linux/ramips/dts/mt7621_elecom_wrc-1167gsxx.dtsi [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+			label = "extender";
+			gpios = <&gpio 43 GPIO_ACTIVE_LOW>;
+			linux,code = <BTN_0>;
+			linux,input-type = <EV_SW>;
+		};
+
+		router {
+			label = "router";
+			gpios = <&gpio 44 GPIO_ACTIVE_LOW>;
+			linux,code = <BTN_0>;
```
Comment (adrianschmutzler) [inline target/linux/ramips/dts/mt7621_elecom_wrc-1167gsxx.dtsi:59]: For buttons linked to the same BTN_0?

**Comment by adrianschmutzler** [comment]
> Both models have a MT7615D and it provides 2.4/5 GHz wifi by single chip. The setup for DBDC (Dual-Band-Dual-Concurrent) mode in device tree is not supported in mt76 driver, so the mode cannot be configured in the dts and the wifi node is temporary state.

So, how are MAC addresses handled there? Will the same address assigned by mtd-mac-address by used twice? Or will this be handled on interface-level (by OpenWrt) with local-bit set? Or won't it work at all based on the limited implementation you reported about?

Is this also the reason why no wifi LED triggers are assigned?

Despite, I assume you didn't put the incremented MAC address for Wifi by accident, as that's really strange again ... ;-)

**Comment by musashino205** [comment]
>So, how are MAC addresses handled there? Will the same address assigned by mtd-mac-address by used twice? Or will this be handled on interface-level (by OpenWrt) with local-bit set? Or won't it work at all based on the limited implementation you reported about?

Currently, only one wifi of MT7615D is recognized in OpenWrt by default. So I've configured the devicetree to use the MAC address for 2.4GHz wifi if there is no correct MAC address in calibration data.
(WRC-1167GS2-B has a MAC address at ```0x4``` in calibration data, however, it's for 5GHz wifi...)
```
root@OpenWrt:/# iwinfo
wlan0     ESSID: unknown
          Access Point: 04:AB:18:58:2F:16
          Mode: Client  Channel: unknown (unknown)
          Tx-Power: 20 dBm  Link Quality: unknown/70
          Signal: unknown  Noise: -84 dBm
          Bit Rate: unknown
          Encryption: unknown
          Type: nl80211  HW Mode(s): 802.11bgnac
          Hardware: 14C3:7615 7615:14C3 [MediaTek MT7615E]
          TX power offset: none
          Frequency offset: none
          Supports VAPs: yes  PHY name: phy0

root@OpenWrt:/# 
```
If DBDC mode is configured manually via sysfs, incremented address may be used for secondary phy.
https://github.com/openwrt/mt76/commit/43db699b1ad87ad446808341b591b62b3d41948d

Note (in stock firmware):

#### WRC-1167GS2-B

- LAN: 04:AB:18:xx:xx:13 (Factory, 0xFFF4 (hex))
- WAN: 04:AB:18:xx:xx:14 (Factory, 0xFFFA (hex))
- 2.4G: 04:AB:18:xx:xx:15 (none)
- 5G: 04:AB:18:xx:xx:16 (Factory, 0x4 (hex))

#### WRC-1167GST2

- LAN: 04:AB:18:xx:xx:07 (Factory, 0xE000 (hex))
- WAN: 04:AB:18:xx:xx:08 (Factory, 0xE006 (hex))
- 2.4G: 04:AB:18:xx:xx:09 (none)
- 5G: 04:AB:18:xx:xx:0A (none)

> Is this also the reason why no wifi LED triggers are assigned?

I haven't configured it because it can't be assigned exactly in dts right now.

File: target/linux/ramips/dts/mt7621_elecom_wrc-1167gsxx.dtsi (hunk 1 above)
Comment (musashino205) [reply target/linux/ramips/dts/mt7621_elecom_wrc-1167gsxx.dtsi:59]: Yes

**Comment by musashino205** [comment]
UPDATE:
- rebase from master

**Comment by musashino205** [comment]
@adschm What's not enough to be merged?

**Comment by adschm** [comment]
@musashino205 I don't have time to build an opinion on this dual-wifi-one-chip issue at the moment.

**Comment by musashino205** [comment]
I'll re-send the patch to the ML, close.

**Pull request closed without merging**